controllers/chat/getChannelDetail.js

```diff
@@ -17,16 +17,32 @@ const getChannelDetail = async (req, res) => {
       message: 'Bad request'
     });
   }
-
+  let {last_fetch_date = null} = req.query;
+  if (last_fetch_date) {
+    last_fetch_date = new Date(last_fetch_date);
+    last_fetch_date = last_fetch_date.toISOString();
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The conversion of `last_fetch_date` to an ISO string after creating a Date object seems unnecessary. JavaScript's Date objects can be compared directly, and the MongoDB query should handle the conversion implicitly. This could simplify the code.
```diff
-    last_fetch_date = new Date(last_fetch_date);
-    last_fetch_date = last_fetch_date.toISOString();
+    last_fetch_date = new Date(last_fetch_date);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->